homeassistant/components/volvooncall.py

```diff
@@ -85,52 +72,79 @@ def setup(hass, config):
 
     def discover_vehicle(vehicle):
         """Load relevant platforms."""
-        state.entities[vehicle.vin] = []
-        for attr, (component, *_) in RESOURCES.items():
-            if (getattr(vehicle, attr + '_supported', True) and
-                    attr in config[DOMAIN].get(CONF_RESOURCES, [attr])):
-                discovery.load_platform(
-                    hass, component, DOMAIN, (vehicle.vin, attr), config)
-
-    def update_vehicle(vehicle):
-        """Receive updated information on vehicle."""
-        state.vehicles[vehicle.vin] = vehicle
-        if vehicle.vin not in state.entities:
-            discover_vehicle(vehicle)
-
-        for entity in state.entities[vehicle.vin]:
-            entity.schedule_update_ha_state()
-
-        dispatcher_send(hass, SIGNAL_VEHICLE_SEEN, vehicle)
-
-    def update(now):
+        state.vehicles.append(vehicle.vin)
+
+        dashboard = vehicle.dashboard(
+            mutable=config[DOMAIN][CONF_MUTABLE],
+            scandinavian_miles=config[DOMAIN][CONF_SCANDINAVIAN_MILES])
+
+        def is_enabled(attr):
+            return attr in config[DOMAIN].get(CONF_RESOURCES, [attr])
+
+        COMPONENTS = {
```

Comment:
This is a constant so we can move it to the module level.

Reply:
✓